### ISSUE TITLE
Exception while rendering the layout editor

### ISSUE THREAD

**Issue by algernon** [keyboardio/Chrysalis#292]
When I use my [own sketch](https://git.madhouse-project.org/algernon/Model01-Sketch), if I connect to the keyboard, I get a white screen, and the following on the console:

```text
Uncaught TypeError: Cannot read property 'components' of null
    at LayoutEditor.render (webpack-internal:///./src/renderer/screens/LayoutEditor.js:361)
    at finishClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14524)
    at updateClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14479)
    at beginWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:15325)
    at performUnitOfWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18144)
    at workLoop (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18184)
    at HTMLUnknownElement.callCallback (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:199)
    at invokeGuardedCallback (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:256)
    at replayUnitOfWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:17431)
/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:16003 The above error occurred in the <LayoutEditor> component:
    in LayoutEditor (created by WithStyles(LayoutEditor))
    in WithStyles(LayoutEditor) (created by Context.Consumer)
    in Unknown (created by App)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by App)
    in main (created by App)
    in LocationProvider (created by App)
    in div (created by App)
    in App (created by WithStyles(App))
    in WithStyles(App) (created by Context.Consumer)
    in Unknown
    in SnackbarProvider

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
logCapturedError @ /home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:16003
/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/@reach/router/index.js:107 Uncaught TypeError: Cannot read property 'components' of null
    at LayoutEditor.render (webpack-internal:///./src/renderer/screens/LayoutEditor.js:361)
    at finishClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14524)
    at updateClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14479)
    at beginWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:15325)
    at performUnitOfWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18144)
    at workLoop (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18184)
    at renderRoot (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18270)
    at performWorkOnRoot (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19159)
    at performWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19071)
    at performSyncWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19045)
/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:16003 The above error occurred in the <LocationProvider> component:
    in LocationProvider (created by App)
    in div (created by App)
    in App (created by WithStyles(App))
    in WithStyles(App) (created by Context.Consumer)
    in Unknown
    in SnackbarProvider

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ /home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:16003
webpack-internal:///./src/renderer/screens/LayoutEditor.js:361 Uncaught (in promise) TypeError: Cannot read property 'components' of null
    at LayoutEditor.render (webpack-internal:///./src/renderer/screens/LayoutEditor.js:361)
    at finishClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14524)
    at updateClassComponent (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:14479)
    at beginWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:15325)
    at performUnitOfWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18144)
    at workLoop (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18184)
    at renderRoot (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:18270)
    at performWorkOnRoot (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19159)
    at performWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19071)
    at performSyncWork (:9080/home/algernon/src/keyboardio/chrysalis/chrysalis-bundle-keyboardio/node_modules/react-dom/cjs/react-dom.development.js:19045)
```

If I add an `if (!focus.device) return null;` to `LayoutEditor`'s `render()` (right after `const focus`), then it times out first, but I'm able to connect the second time.

If I use the factory firmware, built against current Kaleidoscope master - that seems to work.

**Comment by algernon** [comment]
If I revert #267, it works again.

I'm not seeing the connection yet.

**Comment by algernon** [comment]
Fixed by #293
